tests/checkpoint/test_checkpoint.py

```diff
@@ -121,6 +121,8 @@ def test_basic_checkpoint_config_validation(
         ]
     )
 
+    assert len(empty_data_context.list_checkpoints()) == 0
+    empty_data_context.add_checkpoint(**yaml.load(yaml_config_erroneous))
```

Comment:
Two things:
1. I love how defensive you made these tests by adding the assertions before the call you are testing! ❤️ 
2. This variable name caught my eye and raises a question I don't know the answer to. If we pass in that particular "erroneous" config would we expect the call to `add_checkpoint` to work? Perhaps we could think through that together synchronously?

Reply:
Anthony and I discussed my question and decided with the last state of the "erroneous" config we would expect the `add_checkpoint` to succeed.